src/USER-BOCS/MOD_COMPUTE_H

```diff
@@ -0,0 +1,207 @@
+  int size_peratom_cols;    // 0 = vector, N = columns in peratom array
+
+  int local_flag;           // 0/1 if compute_local() function exists
+  int size_local_rows;      // rows in local vector or array
+  int size_local_cols;      // 0 = vector, N = columns in local array
+
+  int extscalar;            // 0/1 if global scalar is intensive/extensive
+  int extvector;            // 0/1/-1 if global vector is all int/ext/extlist
+  int *extlist;             // list of 0/1 int/ext for each vec component
+  int extarray;             // 0/1 if global array is all intensive/extensive
+
+  int tempflag;       // 1 if Compute can be used as temperature
+                      // must have both compute_scalar, compute_vector
+  int pressflag;      // 1 if Compute can be used as pressure (uses virial)
+                      // must have both compute_scalar, compute_vector
+  int pressatomflag;  // 1 if Compute calculates per-atom virial
+  int peflag;         // 1 if Compute calculates PE (uses Force energies)
+  int peatomflag;     // 1 if Compute calculates per-atom PE
+  int create_attribute;    // 1 if compute stores attributes that need
+                           // setting when a new atom is created
+
+  int tempbias;       // 0/1 if Compute temp includes self/extra bias
+
+  int timeflag;       // 1 if Compute stores list of timesteps it's called on
+  int ntime;          // # of entries in time list
+  int maxtime;        // max # of entries time list can hold
+  bigint *tlist;      // list of timesteps the Compute is called on
+
+  int invoked_flag;       // non-zero if invoked or accessed this step, 0 if not
+  bigint invoked_scalar;  // last timestep on which compute_scalar() was invoked
+  bigint invoked_vector;  // ditto for compute_vector()
+  bigint invoked_array;   // ditto for compute_array()
+  bigint invoked_peratom; // ditto for compute_peratom()
+  bigint invoked_local;   // ditto for compute_local()
+
+  double dof;         // degrees-of-freedom for temperature
+
+  int comm_forward;         // size of forward communication (0 if none)
+  int comm_reverse;         // size of reverse communication (0 if none)
+  int dynamic_group_allow;  // 1 if can be used with dynamic group, else 0
+
+  // KOKKOS host/device flag and data masks
+
+  ExecutionSpace execution_space;
+  unsigned int datamask_read,datamask_modify;
+
+  int copymode;
+
+  Compute(class LAMMPS *, int, char **);
+  virtual ~Compute();
+  void modify_params(int, char **);
+  void reset_extra_dof();
+
+  virtual void init() = 0;
+  virtual void init_list(int, class NeighList *) {}
+  virtual void setup() {}
+  virtual double compute_scalar() {return 0.0;}
+  virtual void compute_vector() {}
+  virtual void compute_array() {}
+  virtual void compute_peratom() {}
+  virtual void compute_local() {}
+  virtual void set_arrays(int) {}
+
+// NJD MRD Adding these lines is the only modification needed for USER-BOCS
+  virtual void send_cg_info(int, int, double*, int, double) {}
```

Comment:
This seems unnecessary. Instead of changing the base class `Compute`, why don't you just store `pressure` as `ComputePressureBocs*` in class `FixBocs`? One doesn't work without the other anyway. So you should try to find a compute of type `ComputePressureBocs` and verify if it has a compatible style and do a type cast from `Compute*` to `ComputePressureBocs*`.